### PR TITLE
Add subtree numba-dppy

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
Command:
1. Create: `git subtree add --prefix=numba-dppy  https://github.com/IntelPython/numba-dppy.git main`
2. Pull changes from numba-dppy: `git subtree pull --prefix=numba-dppy https://github.com/IntelPython/numba-dppy.git main`

It could help to make PRs which require synchronous changes to `numba` and `numba-dppy`.

This PR should be merged, not squashed.

### CODE REVIEW

File: numba-dppy/LICENSE [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Copyright (c) 2012, Anaconda, Inc.
```
Comment (diptorupd) [inline numba-dppy/LICENSE]: Should this not be our License file?

File: numba-dppy/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# numba-dppy
+
+## Numba + dpCtl + dpNP = numba-dppy
```
Comment (diptorupd) [inline numba-dppy/README.md:3]: maybe Numba + DPPY + dpCtl + dpNP = numba-dppy :)

File: numba-dppy/LICENSE (hunk 1 above)
Comment (PokhodenkoSA) [reply numba-dppy/LICENSE]: This is license file from numba-dppy. I think it contains an error.

Comment (PokhodenkoSA) [reply numba-dppy/LICENSE]: Updated.

File: numba-dppy/README.md (hunk 2 above)
Comment (PokhodenkoSA) [reply numba-dppy/README.md:3]: @diptorupd please, provide a PR to https://github.com/IntelPython/numba-dppy.git

**Pull request merged**